### PR TITLE
#150507660 mark notifications as read

### PR DESCRIPTION
**Pull request by andela-ojoloko**
#### What does this PR do?
It implements mark of a notification as `read` and `unread` and marking all notifications as` read`.
#### Description of Task to be completed?

- create endpoint `patch api/v1/notifications/:notif_id` to toggle read field of a single notification.
- create enpoint `patch api/v1/notifications` to mark all notifcations as read. 
- update tests 
#### How should this be manually tested?
`pytest -v` 
#### What are the relevant pivotal tracker stories?
 #150507660
#### Questions:
- Do we need a payload to for the endpoint `patch api/v1/notifications` to mark all notifcations as read ?

### CODE REVIEW

File: api/views/notifications.py [hunk 1]
```diff
@@ -15,12 +15,31 @@ def get(self, notif_id=None):
         user_id = g.current_user.user_id
         if notif_id:
             notifications = Notification.query_redis(user_id, notif_id)
-            response = {'status': 'sucess', 'data': notifications}
+            response = {'status': 'success', 'data': notifications}
```
Comment (Mnickii) [inline api/views/notifications.py]: Nice catch 😄

File: api/models.py [hunk 2]
```diff
@@ -317,8 +316,12 @@ def __init__(self, message, category, resource_id, *to, **sender):
     def save(self, redis=True):
         """Save notification in database."""
         if redis:
-            redis_connection.hmset(self.user_id,
-                                   {self.id: json.dumps(self.serialize())})
+            pipeline.hset(self.user_id,
+                                   self.id, json.dumps(self.serialize()))
```
Comment (Mnickii) [inline api/models.py]: I'm concerned about how using pipeline will affect the realtime-ness moving forward. In the case that multiple requests to the server might delay an aspect in the real-time communication. @andela-ojoloko

Comment (andela-ojoloko) [reply api/models.py]: Pipelines are actually faster than making several calls to the `redis` server. A pipeline was used to ensure that all the commands in the pipeline are  processed  or none are processed. This ensures no command in the pipeline is executed just incase the connection to our `redis` db is lost. The pipelines are only executed when we call the `pipeline.execute()` function.  you can read more about pipelines [here](https://redis.io/topics/pipelining)  and  [here](https://github.com/andymccurdy/redis-py#pipelines)

**Comment by Mnickii** [comment]
LGTM

**Comment by Mnickii** [comment]
As for your question, no we don't need a payload because the patch request should update the notifications of the user who is logged in. @andela-ojoloko

**Pull request merged**